File: PlacementCellBackend/Controllers/FeedBack/employeeonstudentsController.cs
using Microsoft.AspNetCore.Mvc;
using PlacementCellBackend.DTOs.EmployeeFeedbackonStudent;
using PlacementCellBackend.Services.Feedback.Interfaces;

namespace PlacementCellBackend.Controllers.FeedBack
{
    [ApiController]
    [Route("api/[controller]")]
    public class EmployeeOnStudentsController : ControllerBase
    {
        private readonly IEmployeeOnStudentService _employeeOnStudentService;

        public EmployeeOnStudentsController(IEmployeeOnStudentService employeeOnStudentService)
        {
            _employeeOnStudentService = employeeOnStudentService;
        }

        [HttpGet]
        public async Task<ActionResult<IEnumerable<EmployeeFeedbackonStudentDtos>>> GetEmployeeOnStudents()
        {
            var employees = await _employeeOnStudentService.GetAllEmployeeOnStudentsAsync();
            return Ok(employees);
        }

        [HttpGet("{id}")]
        public async Task<ActionResult<EmployeeFeedbackonStudentDtos>> GetEmployeeOnStudent(int id)
        {
            var employee = await _employeeOnStudentService.GetEmployeeOnStudentByIdAsync(id);
            if (employee == null)
                return NotFound();
            return Ok(employee);
        }

        [HttpPost]
        public async Task<ActionResult<bool>> PostEmployeeOnStudent(EmployeeFeedbackonStudentCreateDtos employeeOnStudent)
        {
            if (employeeOnStudent == null)
                return BadRequest("Invalid data.");

            var created = await _employeeOnStudentService.CreateEmployeeOnStudentAsync(employeeOnStudent);
            return created;
        }

        [HttpPut("{id}")]
        public async Task<IActionResult> PutEmployeeOnStudent(int id, EmployeeFeedbackonStudentCreateDtos updatedEmployeeOnStudent)
        {


            var success = await _employeeOnStudentService.UpdateEmployeeOnStudentAsync(id, updatedEmployeeOnStudent);
            if (!success)
                return NotFound();

            return NoContent();
        }

        [HttpDelete("{id}")]
        public async Task<IActionResult> DeleteEmployeeOnStudent(int id)
        {
            var success = await _employeeOnStudentService.DeleteEmployeeOnStudentAsync(id);
            if (!success)
                return NotFound();

            return NoContent();
        }
    }
}

File: PlacementicsBackend/src/main/java/com/example/placementicsbackend/controllers/FeedBack/employeeonstudentsController.java
package com.example.placementicsbackend.controllers.FeedBack;

import com.example.placementicsbackend.dtos.EmployeeFeedbackonStudent.EmployeeFeedbackOnStudentCreateDto;
import com.example.placementicsbackend.dtos.EmployeeFeedbackonStudent.EmployeeFeedbackOnStudentDto;
import com.example.placementicsbackend.services.Feedback.Interfaces.IEmployeeOnStudentService;
import lombok.RequiredArgsConstructor;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping("/api/employee-on-students")
@RequiredArgsConstructor
public class employeeonstudentsController {

    private final IEmployeeOnStudentService employeeOnStudentService;

    @GetMapping
    public ResponseEntity<List<EmployeeFeedbackOnStudentDto>> getEmployeeOnStudents() {
        return ResponseEntity.ok(employeeOnStudentService.getAllEmployeeOnStudents());
    }

    @GetMapping("/{id}")
    public ResponseEntity<EmployeeFeedbackOnStudentDto> getEmployeeOnStudent(@PathVariable Integer id) {
        EmployeeFeedbackOnStudentDto employee = employeeOnStudentService.getEmployeeOnStudentById(id);
        if (employee == null) {
            return ResponseEntity.notFound().build();
        }
        return ResponseEntity.ok(employee);
    }

    @PostMapping
    public ResponseEntity<Boolean> postEmployeeOnStudent(
            @RequestBody EmployeeFeedbackOnStudentCreateDto employeeOnStudent) {
        if (employeeOnStudent == null) {
            return ResponseEntity.badRequest().body(false);
        }
        return ResponseEntity.status(HttpStatus.CREATED)
                .body(employeeOnStudentService.createEmployeeOnStudent(employeeOnStudent));
    }

    @PutMapping("/{id}")
    public ResponseEntity<Void> putEmployeeOnStudent(
            @PathVariable Integer id,
            @RequestBody EmployeeFeedbackOnStudentCreateDto updatedEmployeeOnStudent) {
        if (!employeeOnStudentService.updateEmployeeOnStudent(id, updatedEmployeeOnStudent)) {
            return ResponseEntity.notFound().build();
        }
        return ResponseEntity.noContent().build();
    }

    @DeleteMapping("/{id}")
    public ResponseEntity<Void> deleteEmployeeOnStudent(@PathVariable Integer id) {
        if (!employeeOnStudentService.deleteEmployeeOnStudent(id)) {
            return ResponseEntity.notFound().build();
        }
        return ResponseEntity.noContent().build();
    }
}
